cmake/Modules/LocalDav1d.cmake

```diff
@@ -0,0 +1,106 @@
+        if(NOT CMAKE_SYSTEM_PROCESSOR STREQUAL CMAKE_HOST_SYSTEM_PROCESSOR)
+            string(TOLOWER "${CMAKE_SYSTEM_NAME}" cross_system_name)
+            if(CMAKE_C_BYTE_ORDER STREQUAL "BIG_ENDIAN")
+                set(cross_system_endian "big")
+            else()
+                set(cross_system_endian "little")
+            endif()
+            if(CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
+                set(cross_system_processor "aarch64")
+            else()
+                set(cross_system_processor "${CMAKE_SYSTEM_PROCESSOR}")
+            endif()
+            if(CMAKE_OSX_DEPLOYMENT_TARGET)
+                set(cross_osx_deployment_target "-mmacosx-version-min=${CMAKE_OSX_DEPLOYMENT_TARGET}")
+            endif()
+
+            set(CROSS_FILE "${PROJECT_BINARY_DIR}/crossfile-apple.meson")
+            configure_file("cmake/Meson/crossfile-apple.meson.in" "${CROSS_FILE}")
+        endif()
+    endif()
+
+    if(CROSS_FILE)
+        set(EXTRA_ARGS "--cross-file=${CROSS_FILE}")
+    endif()
+
+    set(install_prefix "${binary_dir}/install.libavif")
+
+    file(MAKE_DIRECTORY ${install_prefix}/include)
+
+    ExternalProject_Add(
+        dav1d
+        GIT_REPOSITORY https://code.videolan.org/videolan/dav1d.git
+        SOURCE_DIR "${source_dir}"
+        PREFIX "${binary_dir}"
+        INSTALL_DIR "${install_prefix}"
+        LIST_SEPARATOR |
+        GIT_TAG ${AVIF_LOCAL_DAV1D_TAG}
+        GIT_SHALLOW ON
+        UPDATE_COMMAND ""
+        CONFIGURE_COMMAND
+            ${CMAKE_COMMAND} -E env "PATH=${PATH}" ${MESON_EXECUTABLE} setup --buildtype=release --default-library=static
+            --prefix=<INSTALL_DIR> --libdir=lib -Denable_asm=true -Denable_tools=false -Denable_examples=false
+            -Denable_tests=false ${EXTRA_ARGS} <SOURCE_DIR>
+        BUILD_COMMAND ${CMAKE_COMMAND} -E env "PATH=${PATH}" ${NINJA_EXECUTABLE} -C <BINARY_DIR>
+        INSTALL_COMMAND ${CMAKE_COMMAND} -E env "PATH=${PATH}" ${NINJA_EXECUTABLE} -C <BINARY_DIR> install
+        BUILD_BYPRODUCTS <INSTALL_DIR>/lib/libdav1d.a
+    )
+
+    add_library(dav1d::dav1d STATIC IMPORTED)
+    set_target_properties(dav1d::dav1d PROPERTIES IMPORTED_LOCATION ${install_prefix}/lib/libdav1d.a AVIF_LOCAL ON)
+    target_include_directories(dav1d::dav1d INTERFACE "${install_prefix}/include")
+    target_link_directories(dav1d::dav1d INTERFACE ${install_prefix}/lib)
+    add_dependencies(dav1d::dav1d dav1d)
+
+    if(UNIX AND NOT APPLE)
+        target_link_libraries(dav1d::dav1d INTERFACE ${CMAKE_DL_LIBS}) # for dlsym
+    endif()
+
+    set(DAV1D_FOUND ON PARENT_SCOPE)
+    set(DAV1D_LIBRARY dav1d::dav1d PARENT_SCOPE)
+    set(DAV1D_LIBRARY_DIRS ${install_prefix}/lib PARENT_SCOPE)
+    set(DAV1D_INCLUDE_DIR ${install_prefix}/include PARENT_SCOPE)
+    set(DAV1D_VERSION ${AVIF_LOCAL_DAV1D_TAG} PARENT_SCOPE)
+    set_target_properties(dav1d::dav1d PROPERTIES FOLDER "ext/dav1d")
+endfunction()
+
+avif_build_local_dav1d()
```

Comment:
To scope the variables. Otherwise it would pollute the global namespace with a bunch of dav1d-specific variables. What I've seen in the cmake source itself as an alternative is to create macros that clear and then unset the variables set in a block of code.